### PR TITLE
Usda elderly disabled

### LINKED ISSUE #236: make USDA is_usda_disabled flag and is_usda_elderly flag

### PR DESCRIPTION
**Pull request by nmrodelo**
Add variables for disabled and elderly status as per USDA rules
fixes #236

### CODE REVIEW

File: openfisca_us/variables/finance/benefit/snap.py [hunk 1]
```diff
@@ -156,9 +156,15 @@ def formula(spm_unit, period, parameters):
         )
 
 
-class is_disabled_or_elderly_for_snap(Variable):
+class usda_disabled(Variable):
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/benefit/snap.py:159]:
```suggestion
class is_usda_disabled(Variable):
```

File: openfisca_us/variables/finance/benefit/snap.py [hunk 2]
```diff
@@ -156,9 +156,15 @@ def formula(spm_unit, period, parameters):
...
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    documentation = "Indicates that a person is defined as disabled based on the USDA definition"
+
 
+class usda_elderly(Variable):
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/benefit/snap.py:166]:
```suggestion
class is_usda_elderly(Variable):
```

**Review by MaxGhenis — COMMENTED** [review]
Thanks @nmrodelo but I think #234 was for the parameter not the variable. Could you add that as well as a formula for `is_usda_elderly` referencing the parameter? Or unlink the issue and save it for another PR.

**Comment by nmrodelo** [comment]
> Thanks @nmrodelo but I think #234 was for the parameter not the variable. Could you add that as well as a formula for `is_usda_elderly` referencing the parameter? Or unlink the issue and save it for another PR.

Yeah you're right, unlinked it. I have the params saved in a gist currently:
https://gist.github.com/nmrodelo/e9d6a56c535fd810807ebd60fa1a7bf3

These params are definitely not ready to pull, gonna need to discuss the in the meeting today.

File: openfisca_us/variables/finance/benefit/snap.py [hunk 3]
```diff
@@ -156,9 +156,15 @@ def formula(spm_unit, period, parameters):
         )
 
 
-class is_disabled_or_elderly_for_snap(Variable):
+class is_usda_disabled(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    documentation = "Indicates that a person is defined as disabled based on the USDA definition"
```
Comment (nikhilwoodruff) [inline openfisca_us/variables/finance/benefit/snap.py:163]: Could we add a label? And below

Comment (nmrodelo) [reply openfisca_us/variables/finance/benefit/snap.py:163]: and below?

Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/benefit/snap.py:163]: Just in the other variables too (`is_usda_elderly` etc.). Thanks!

**Comment by nmrodelo** [comment]
Need to update old combined is_elderly_or_disabled references as part of this pr as well

File: openfisca_us/parameters/benefit/snap/usda_elderly_and_disabled [hunk 4]
```diff
@@ -0,0 +1,32 @@
+description: USDA Definitions for Elderly and Disabled Individuals
+
+elderly:
+  age:
+    2021-01-01: 60
+  metadata:
+    unit: years
+    
+disabled:
+  federal_disability_or_blindess_receipt:
```
Comment (MaxGhenis) [inline openfisca_us/parameters/benefit/snap/usda_elderly_and_disabled:10]: You can replace the bools with a list like this:
```yaml
2021-01-01:
  - federal_disability_or_blindess_receipt
  - state_disability_or_blindess_receipt
  - gov_agency_permanent_disability
  - railroad_retirement_and_medicare
  - ssi_disabled
  - permanently_disabled_veteran
  - surviving_spouse_of_disabled_veteran
  - surviving_child_of_disabled_veteran
```

File: openfisca_us/parameters/benefit/snap/usda_elderly_and_disabled [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  surviving_spouse_of_disabled_veteran: 
+    2021-01-01: True
+  surviving_child_of_disabled_veteran: 
+    2021-01-01: True
+  metadata:
+    unit: program participation
+ 
+metadata: 
+  reference:
+      - title: 'SNAP Special Rules for the Elderly or Disabled'
```
Comment (MaxGhenis) [inline openfisca_us/parameters/benefit/snap/usda_elderly_and_disabled:31]: too much indent

**Review by MaxGhenis — COMMENTED** [review]
Given the new structure (which is creating the merge conflict), I'd suggest moving the parameter to `parameters/usda/elderly.yaml` and `disabled.yaml`. Also move this non-SNAP-specific logic into `variables/usda/usda.py`.

I know I previously said it'd be OK to split off the formula into a new PR, but I think it'd be helpful here so that we can test the new parameters and variables here. Please add formulas to the elderly and disabled variables along with some unit tests for them.

**Pull request closed without merging**